### ISSUE TITLE
Size of function_id within Action Space

### ISSUE THREAD

**Issue by hagopi1611** [LunarEngineer/MentalGymnastics#4]
In mentalgym.py -> self.action_space:

We can leave function_id spaces.Discrete(self.number_functions) as is,

where number_functions should be an upper bound on the total number of atomic + composed functions that we can have in our function bank.  We only need to add a persistent variable in the agent's scope that lets it know the total number of functions available to it, which will be the *current* number of functions in the function bank.  The agent will then restrict his action to select  only from available functions.

In other words, we don't need to grow the size of the number of actions the agent can take over time.
